src/goto-instrument/cover.cpp

```diff
@@ -33,22 +33,48 @@ class basic_blockst
   {
     bool next_is_target=true;
     unsigned block_count=0;
+    unsigned current_block=0;
 
     forall_goto_program_instructions(it, _goto_program)
     {
       if(next_is_target || it->is_target())
-        block_count++;
+      {
+        // We increase the block number only if this block
+        // is not a continuation of a previous block through
```

Comment:
maybe simplify / rephrase "We keep the block number if it is a continuation ... ; otherwise we increase the block number"